packages/rum/src/boot/recorderApi.spec.ts

```diff
@@ -104,10 +107,22 @@ describe('makeRecorderApi', () => {
     it('do not start recording if worker fails to be instantiated', () => {
       setupBuilder.build()
       rumInit(DEFAULT_INIT_CONFIGURATION)
-      getDeflateWorkerSingletonSpy.and.returnValue(undefined)
+      loadDeflateWorkerSingletonSpy.and.callFake((callback) => callback(undefined))
       recorderApi.start()
       expect(startRecordingSpy).not.toHaveBeenCalled()
     })
+
+    it('does not start recording multiple times if restarted before worker is ready', () => {
+      setupBuilder.build()
+      rumInit(DEFAULT_INIT_CONFIGURATION)
+      loadDeflateWorkerSingletonSpy.and.callFake(noop)
+      recorderApi.start()
+      recorderApi.stop()
+      loadDeflateWorkerSingletonSpy.calls.mostRecent().args[0]({} as DeflateWorker)
```

Comment:
the intent is a bit hard to follow, what about using a stub with an API explaining what is happening